src/farbtastic.js

```diff
@@ -490,7 +490,7 @@ $._farbtastic = function (container, options) {
   };
 
   // Parse options.
-  if (!options.callback) {
+  if (!options.callback && !options.width) {
     options = { callback: options };
```

Comment:
In my case i have to initialize script with empty callback, set color and then link to callback. There is no way to set starting color if i don't use input field and i have to initialize with options because you can't set width otherwise. Maybe there should be a color option then?